### PR TITLE
Remove declaration of friendship for Executor

### PR DESCRIPTION
**Pull request by jacquelinekay**
This is to make adding allocator templates to stdlib structures in the executor easier.

### CODE REVIEW

File: rclcpp/include/rclcpp/callback_group.hpp [hunk 1]
```diff
@@ -58,6 +53,38 @@ class CallbackGroup
   : type_(group_type), can_be_taken_from_(true)
   {}
 
+  std::vector<subscription::SubscriptionBase::WeakPtr> &
+  get_subscription_ptrs()
```
Comment (dirk-thomas) [inline rclcpp/include/rclcpp/callback_group.hpp]: Can these methods be `const` (and return a const reference)?

Same below.

**Comment by wjwwood** [comment]
That looks like a clean change. The reason I originally did the friend pattern was that I felt the extra functions crowded the user facing API, but I think the result from the pr is still pretty clean. I may try to continue to hide these interfaces when I move `rclcpp` to a library, but in a way where the interfaces are still public (no need for friend) but they are somehow separate from the more normal user functions like `publish`, `get_topic_name`, `get_type`, etc. This also has to be done (in some related fashion) for `Node` being a friend of several classes.

But for now, this gets a +1 from me.

**Comment by jacquelinekay** [comment]
Jenkins after fixing style and an issue with timers that was causing test failures:

http://ci.ros2.org/job/ros2_batch_ci_linux/476/
http://ci.ros2.org/job/ros2_batch_ci_osx/378/
http://ci.ros2.org/job/ros2_batch_ci_windows/527/

**Comment by esteve** [comment]
+1

**Pull request merged**